pkg/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -306,3 +306,21 @@ func (d *dnsRecord) valuesDontMatchDNSRecord() bool {
 func (d *dnsRecord) lastOperationNotSuccessful() bool {
 	return d.dnsRecord.Status.LastOperation != nil && d.dnsRecord.Status.LastOperation.State != gardencorev1beta1.LastOperationStateSucceeded
 }
+
+func (d *dnsRecord) statusIsOutdatedOrTimestampIsInvalid() bool {
```

Comment:
I think this needs more explanation why this check is required. Can you add a comment?

Reply:
Thanks for the feedback! I did the change. I also think that once this change is rolled out then this condition will become obsolete (at least I cannot think of a way to enter this scenario again). I am not sure if it also makes sense to leave a TODO and remove this in a future release 🤔

Reply:
The check will remain valid, not sure if we are going to need it in the future though